lib/oelite/meta/meta.py

```diff
@@ -436,15 +436,18 @@ def dump_var(self, key, o=sys.__stdout__, pretty=True, dynvars=[],
             expand = FULL_EXPANSION
         if not expand and func != "python":
             expand = OVERRIDES_EXPANSION
-        val = self.get(key, expand)
+        val,deps = self._get(key, expand)
 
         if not val:
             return 0
 
         val = str(val)
 
-        for dynvar_name, dynvar_val in dynvars:
-            val = string.replace(val, dynvar_val, dynvar_name)
+        if deps:
+            for dynvar_name, dynvar_token, dynvar_val in dynvars:
+                if dynvar_name not in deps:
+                    continue
+                val = string.replace(val, dynvar_val, dynvar_token)
```

Comment:
As I write in the commit message, having the string '${TOPDIR}/relative/path/recipe.sig' mixed into the hash doesn't really provide anything which isn't already accounted for by other variables which do get hashed. Moreover, even if somebody decides to do 'SIGNATURE_FILE = "${PN}.something_else_just_because_im_special", what is really important for hashing is the signatures contained in that file.

The patch adding these variables to builtin_nohash is a necessary followup to the "safer/saner" dynvar handling.